docs/source/notes/extending.rst

```diff
@@ -854,6 +853,62 @@ useful to confirm that a function that isn't present in the dictionary returned
+
+Some important implication of this implementation are:
+- This code runs "below all features" it is thus only responsible, like a regular backend, to generate the output value of each Tensor (and can, and should, ignore all advanced features like autograd, autocast, etc).
+- If any high level feature implement a give function without redispatching, it will never reach the Python key and so the ``__torch_dispatch__`` callback will never be triggered. This happens in particular for CompositeImplicitAutograd function (composite functions whose autograd formula is specified implicitly by the other native ops it calls) which are evaluated at the Autograd level without redispatching.
+- When calling back to python and when wrapping the results, the usual conversions are used as some objects cannot be represented in python (undefined Tensors for example become None).
+- To have an easy to use python object representing a given native function, they are lazily populated at ``torch.ops.{namespace}.{func_name}.{overload_name}``. The ``func`` object given to ``__torch_dispatch__`` is always an entry from this namespace. This namespace can be used to directly call native ops and bypass the usual python API and binding code.
+
+
+In a similar way where ``__torch_function__`` is able to interpose on all of torch's python API and Tensor methods, ``__torch_dispatch__`` is able to intercept all calls into the aten native API. Note that all methods on Tensors are converted into function calls before entering the dispatcher and thus will appear as function calls here: ``torch.add(a, 2)`` and ``a + 2`` will lead to exactly the same aten call.
+Most of these function are defined in ``native_functions.yaml`` which specifies the properties of these functions as well as their backend implementation. Their registration is then generated via codegen alongside some of the feature registrations.
+Some more exotic functions or features are also registered in other places in the c++ codebase or in user-defined c++ extensions.
+
+It is also possible to add new native functions using :mod:`torch.library`. This python feature allows to define and/or add new implementations to native functions. This can be used to add missing kernels, replace existing ones or define brand new native functions.
```

Comment:
```suggestion
It is also possible to add `new` native functions using :mod:`torch.library`. This python feature allows defining and/or adding new implementations to native functions. This can be used to add missing kernels, replace existing ones or define brand new native functions.
```
I'm guessing "new" is the keyword here? Is the advisory to use torch_dispatch to interpose on the C++ level, torch_function on the Python level _for existing ops_ and then to use torch.library to add missing implementations?  This is likely gonna get confusing with custom ops lol, will you eventually be addressing that too?

Reply:
CustomOp is a layer on top of torch.library to make it easier to use. So yes that will be handled in a follow up.